src/controller/CHIPDevice.h

```diff
@@ -369,6 +369,11 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     void OperationalCertProvisioned();
     bool IsOperationalCertProvisioned() const { return mDeviceOperationalCertProvisioned; }
+    void MarkOperationalCertProvisioned()
+    {
+        mDeviceOperationalCertProvisioned = true;
+        mState                            = ConnectionState::NotConnected;
+    }
```

Comment:
Why is this needed? It's not called

Reply:
It's used for working around assumed comissioning. The controller example is currently in inner projects.

The usage will be like:
```
        SecureSessionMgr &sessionMgr = GetGlobalSessionManager();
        chip::Messaging::ExchangeManager &exchangeMgr = GetGlobalExchangeManager();
        chip::Controller::ControllerDeviceInitParams params = {nullptr,          &sessionMgr,
                                                               &exchangeMgr,     nullptr,
                                                               nullptr,          &mOperationalCredentialSet,
                                                               mCredentialIndex, &GetGlobalSessionIDAllocator(),
                                                               nullptr};
        mDevice.Init(params, CHIP_PORT, mPeerNodeId, mPeerAddr, mFabricIndex);
        mDevice.MarkOperationalCertProvisioned();
```
As to refactoring, my idea is to make CASE peer a seperate class and decouple it with commissiong and persistence. Does it look fine to you?

Reply:
Adding @pan-apple as the origin author of the code.